services/basic_types.proto

```diff
@@ -1592,3 +1592,65 @@ message TokenAllowance {
      */
     int64 amount = 4;
 }
+
+/**
+ * A granted allowance of hbar transfers for a spender relative to the owner account.
+ */
+message GrantedCryptoAllowance {
+    /**
+     * The account ID of the spender of the hbar allowance.
+     */
+    AccountID spender = 1;
+
+    /**
+     * The amount of the spender's allowance in tinybars.
+     */
+    int64 amount = 2;
+}
+
+/**
+ * A granted allowance of non-fungible token transfers for a spender relative to the owner account.
+ */
+message GrantedNftAllowance {
+    /**
+     * The token that the allowance pertains to.
+     */
+    TokenID tokenId = 1;
```

Comment:
Should we use `token_id` here? (And for `serial_numbers`, `approved_for_all`, etc below.)

Reply:
@tinker-michaelj  Converted to snake case